scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java

```diff
@@ -3227,35 +3230,37 @@ private boolean isLabelValid(String label) {
     public List<JobLabelInfo> newLabels(List<String> labels, String username)
             throws LabelConflictException, LabelValidationException {
         List<JobLabelInfo> jobLabelsInfo = new LinkedList<>();
-        for (String label : labels) {
+        Set<String> uniqueLabels = new HashSet<>(labels);
```

Comment:
you should use linkedhashset also here to preserve the order

Reply:
OK done.